File: playwright/pages/LoginPage.js
import { BasePage } from "./BasePage.js";
import { ADMIN_EMAIL, ADMIN_PASSWORD } from "../config/config.js";

export class LoginPage extends BasePage {
  async openLogin() {
    await this.open("/login");
    await this.el("login-form");
  }

  async fillCredentials(email, password) {
    await this.fillInput("login-email", email);
    await this.fillInput("login-password", password);
  }

  async submit() {
    await this.testId("login-submit").click();
  }

  async login(email, password) {
    await this.fillCredentials(email, password);
    await this.submit();
  }

  async loginWithDefaults() {
    await this.login(ADMIN_EMAIL, ADMIN_PASSWORD);
  }

  async hasValidationError(text) {
    return this.page.getByText(text).isVisible();
  }

  async emailValue() {
    return this.testId("login-email").inputValue();
  }

  async passwordValue() {
    return this.testId("login-password").inputValue();
  }

  async isSubmitDisabled() {
    return this.testId("login-submit").isDisabled();
  }
}

File: src/test/java/com/automation/um/pages/LoginPage.java
package com.automation.um.pages;

import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {
  public LoginPage(WebDriver driver) {
    super(driver);
  }

  public void openLogin() {
    open("/login");
    el("login-form");
  }

  public void fillCredentials(String email, String password) {
    fillInput("login-email", email);
    fillInput("login-password", password);
  }

  public void submit() {
    el("login-submit").click();
  }

  public void login(String email, String password) {
    fillCredentials(email, password);
    submit();
  }

  public void loginWithDefaults() {
    login("admin@acme.test", "admin123");
  }

  public boolean hasValidationError(String text) {
    return driver.getPageSource().contains(text);
  }

  public String emailValue() {
    return el("login-email").getAttribute("value");
  }

  public String passwordValue() {
    return el("login-password").getAttribute("value");
  }

  public boolean isSubmitDisabled() {
    return !el("login-submit").isEnabled();
  }
}
